components/date-picker/date-picker.component.ts

```diff
@@ -507,8 +508,12 @@ export class NzDatePickerComponent implements OnInit, OnChanges, OnDestroy, Afte
     event.preventDefault();
     event.stopPropagation();
 
-    this.datePickerService.initValue(true);
-    this.datePickerService.emitValue$.next();
+    // It will fix the issue when in some cases handler will be called runOutsideAngular.
+    // This issue occurs from nz-icon directive because of triggered runOutsideAngular.
+    this.zone.run(() => {
+      this.datePickerService.initValue(true);
+      this.datePickerService.emitValue$.next();
+    });
```

Comment:
Is it possible to use `ChangeDetectorRef.markForCheck` instead of `NgZone.run()`?( `zone.run()` will cause change detection on the whole application).

Reply:
> Is it possible to use `ChangeDetectorRef.markForCheck` instead of `NgZone.run()`?( `zone.run()` will cause change detection on the whole application).

Hi, markForCheck doesn't help here because the current view is rendering outside zone (because of the `nz-icon` directive).

The default `click listener` also globally run inside the zone, it's the same behaviour.